apps/prairielearn/src/pages/instructorAssessmentStatistics/instructorAssessmentStatistics.ts

```diff
@@ -18,53 +25,69 @@ const setFilenames = function (locals) {
     locals.course_instance,
     locals.course,
   );
-  locals.scoreStatsCsvFilename = prefix + 'score_stats.csv';
-  locals.durationStatsCsvFilename = prefix + 'duration_stats.csv';
-  locals.statsByDateCsvFilename = prefix + 'scores_by_date.csv';
+
+  const filenames: Filenames = {
+    scoreStatsCsvFilename: prefix + 'score_stats.csv',
+    durationStatsCsvFilename: prefix + 'duration_stats.csv',
+    statsByDateCsvFilename: prefix + 'scores_by_date.csv',
+  };
```

Comment:
I can't comment on this directly, but, I'd write it this function like this:

```ts
function getFilenames(locals: Record<string, any>): Filenames {
  const prefix = assessmentFilenamePrefix(
    locals.assessment,
    locals.assessment_set,
    locals.course_instance,
    locals.course,
  );

  return {
    scoreStatsCsvFilename: prefix + 'score_stats.csv',
    durationStatsCsvFilename: prefix + 'duration_stats.csv',
    statsByDateCsvFilename: prefix + 'scores_by_date.csv',
  };
};
```

Commentary:

- Let's call it `getFilenames` instead; it's not doing any setting anymore.
- When a function has an explicit return type, there's no need to add an explicit type to the return value.

Reply:
That all makes sense. Wrote the function like above in [4dad954](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/4dad954294139e1ccb0018548e102663ab43c2df).